handout/modules/shared-di-tree.md

```diff
@@ -0,0 +1,62 @@
+# Sharing the Same Dependency Injection Tree
+
+So far our problem is that we are creating two instances of the same services in different levels of the DI tree and that the instance that is created in the lower branch of the tree, is shadowing the one defined at the root level. The solution? To avoid creating a second instance in a lower level of the DI tree for the lazy loaded module and only use the service instance registered at the root of the tree.
```

Comment:
- Break up the sentence `So far our problem is that we are creating two instances of the same services in different levels of the DI tree and that the instance that is created in the lower branch of the tree, is shadowing the one defined at the root level` to `...in different levels of the DI tree. The instance created in the lower branch of the tree is shadowing the one defined at the root level`